src/model.ts

```diff
@@ -1,3 +1,44 @@
-export interface User {
-  name: string;
+export interface ValidationResult {
+  key?: string;
+  type: string;
+  succeeded: boolean;
+  errorMessage: string;
+}
+
+export const createDefaultValidationResult = (): ValidationResult => ({
+  key: '',
+  type: '',
+  succeeded: true,
+  errorMessage: '',
+});
+
+// Pending on ValidationEngine
+type ValidationResultSyncAsync = ValidationResult | Promise<ValidationResult>;
+
+export type FieldValidationFunction = (
+  value: any,
+  values: any,
```

Comment:
`values` and `customParams` params should be optional, isn' it?

Reply:
You are right, updated